Add BodyTypesController

Nikitosique opened the pull request:
Add BodyTypesController, add getAll.html, update BodyTypeDAO

Nikitosique committed:
Add BodyTypesController

Add BodyTypesController, add getAll.html, update BodyTypeDAO

AndryLat commented on src/main/java/dev/andrylat/carsharing/controllers/BodyTypesController.java:
```diff
@@ -0,0 +1,33 @@
+                         @RequestParam(name = "pageSize", defaultValue = "10") int pageSize,
+                         Model model) {
+        model.addAttribute("bodyTypes", bodyTypeDAO.getAll(pageNumber, pageSize));
+        model.addAttribute("recordsNumber", bodyTypeDAO.getRecordsNumber());
```
Всмысле?! А зачем мы тогда сервисный слой делали? Чтобы ты напрямую с дао слоем работал, без каких либо проверко и т.д.?!

Nikitosique committed:
Add service class for BodyType entity

Add service class for BodyType entity, add Validators for query and path parameters,
add ExceptionHandler, fix controller class for BodyType entity,
add views for BodyTypesController's methods, add html-pages error info.

AndryLat commented on src/main/java/dev/andrylat/carsharing/controllers/BodyTypesController.java:
```diff
@@ -0,0 +1,40 @@
+@RequestMapping("/bodytypes")
+public class BodyTypesController {
+
+    private final BodyTypeServiceImpl bodyTypeServiceImpl;
```
Почему не интерфейс?

Nikitosique committed:
Fix bugs

Fix bugs in BodyTypesController
